services/dynamic/json-path.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * @module
+ */
+
+'use strict'
+
+const Joi = require('@hapi/joi')
+const jp = require('jsonpath')
+const { renderDynamicBadge, errorMessages } = require('../dynamic-common')
+const { InvalidParameter, InvalidResponse } = require('..')
+
+/**
+ * Dynamic service class factory which wraps {@link module:core/base-service/base~BaseService} with support of {@link https://jsonpath.com/|JSONPath}.
+ *
+ * @param {Function} superclass class to extend
+ * @returns {Function} wrapped class
+ */
+module.exports = superclass =>
+  class extends superclass {
+    static get category() {
+      return 'dynamic'
+    }
+
+    static get defaultBadgeData() {
+      return {
+        label: 'custom badge',
+      }
+    }
+
+    /**
+     * Request data from an upstream API, transform it to JSON and validate against a schema
+     *
+     * @param {object} attrs Refer to individual attrs
+     * @param {Joi} attrs.schema Joi schema to validate the response transformed to JSON
+     * @param {string} attrs.url URL to request
+     * @param {object} [attrs.errorMessages={}] Key-value map of status codes
+     *    and custom error messages e.g: `{ 404: 'package not found' }`.
+     *    This can be used to extend or override the
+     *    [default](https://github.com/badges/shields/blob/master/services/dynamic-common.js#L8)
+     * @returns {object} Parsed response
+     */
+    async _getData({ schema, url, errorMessages }) {
```

Comment:
Could we use the `fetch` naming convention here?

Reply:
What exactly do you mean? Would you like to change `_getData` to `fetch`? Should I change parameters?

Reply:
Sorry I should have been more clear, but yes I was suggesting changing the function name from `_getData` to `fetch`. I know `_getData` is consistent with what was being used before in the json/yaml service classes, but I feel like we use the function name `fetch` in other service classes for this responsibility and I think it'd be good to do so in these dynamic badge services for consistency

Reply:
Thanks for clarification. Done.